rust/src/main.rs

```diff
@@ -73,15 +72,62 @@ fn main() -> anyhow::Result<()> {
         }
     }));
 
+    let enabled_flags: HashSet<_> = config.features.file_flags.iter().cloned().collect();
+
     for test_case in TEST_CASES {
-        if let Some(sc) = &test_case.syscall {
-            if !enabled_features.contains(sc) {
-                println!(
-                    "skipped {}: please add it to the configuration file if you want to run it",
-                    sc
-                );
-                continue;
-            }
+        //TODO: There's probably a better way to do this...
+        let mut should_skip = false;
+
+        let mut message = None;
+
+        let features = test_case
+            .required_features
+            .iter()
+            .cloned()
+            .collect::<HashSet<_>>();
+        let missing_features = features.difference(&enabled_features);
+        if missing_features.clone().count() > 0 {
```

Comment:
Why the `clone()`?

Reply:
Since it's an iterator, the `count()` method would move the value and therefore wouldn't allow using it after.

Reply:
Should it be better to convert it to a `Vec`?

Reply:
That would make sense.  I'm honestly not sure which would be faster, but collecting it into a Vec would be more readable.